exporter/kafkaexporter/kafka_exporter.go

```diff
@@ -332,3 +326,24 @@ func metadataToHeaders(ctx context.Context, keys []string) []sarama.RecordHeader
 	}
 	return headers
 }
+
+func wrapKafkaProducerError(err error) error {
+	var prodErr sarama.ProducerErrors
+	if !errors.As(err, &prodErr) || len(prodErr) == 0 {
+		return err
+	}
+
+	var areConfigErrs bool
+	var confErr sarama.ConfigurationError
+	for _, producerErr := range prodErr {
+		if areConfigErrs = errors.As(producerErr.Err, &confErr); !areConfigErrs {
+			break
+		}
+	}
+
+	if areConfigErrs {
+		return consumererror.NewPermanent(confErr)
+	}
+
+	return kafkaErrors{len(prodErr), prodErr[0].Err.Error()}
```

Comment:
This feels like we are loosing some information performing this. I might follow up in a seperate PR.